.github/workflows/update_sources.yml

```diff
@@ -9,6 +9,11 @@ on:
       version:
         description: 'Target Version without (-SNAPSHOT)'
         required: true
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
```

Comment:
Github CI uses concurrency groups to determine concurrency. So, we need to create a group name that is unique to each PR and each workflow in that PR. However, when it comes to tags and master, `github.event.pull_request.number` will not exist. `||` allow us to specify an alternative, which in this case is the fully formed `ref`.